pkg/log/rotate.go

```diff
@@ -4,31 +4,38 @@
 package log
 
 import (
+	"bufio"
+	"compress/gzip"
 	"errors"
 	"fmt"
-	"github.com/newrelic/infrastructure-agent/pkg/disk"
+	"io"
 	"os"
 	"path/filepath"
 	"strings"
 	"sync"
 	"time"
+
+	"github.com/newrelic/infrastructure-agent/pkg/disk"
 )
 
 var rLog = WithComponent("LogRotator")
 
-// defaultDatePattern used to generate filename for the rotated file.
-const defaultDatePattern = "YYYY-MM-DD_hh-mm-ss"
-
-var (
-	// ErrFileNotOpened is returned when an operation cannot be performed because the file is not opened.
-	ErrFileNotOpened = errors.New("cannot perform operation, file is not opened")
+const (
+	// defaultDatePattern used to generate filename for the rotated file.
+	defaultDatePattern = "YYYY-MM-DD_hh-mm-ss"
+	// filePerm specified the permissions while opening a file.
+	filePerm = 0o666
```

Comment:
`filePerm = 0666` ☝️😉   ?

Reply:
gofumpt does this, otherwise linter fails